examples/rust/Earthfile

```
@@ -0,0 +1,44 @@
+FROM rust:1.49
+WORKDIR /example-rust
+
+plan:
+    RUN cargo install cargo-chef --version 0.1.11
+    
+    # Compute a lock-like file for our project
+    RUN cargo chef prepare --recipe-path recipe.json
+    SAVE ARTIFACT recipe.json recipe.json
+deps:
+    FROM +plan
+
+    COPY . .
+    COPY +plan/recipe.json .
+
+    RUN cargo chef cook --release --recipe-path recipe.json
+    SAVE ARTIFACT /usr/local/cargo cargo
+    SAVE ARTIFACT target target
+
+build:
+    FROM +deps
+
+    # Copy over the cached dependencies
+    COPY +deps/target .
+    COPY +deps/cargo /usr/local/cargo
+
+    ENV PORT 9091
+    RUN cargo build --release --bin example-rust
+    
+    SAVE ARTIFACT target/release/example-rust example-rust
+
+docker:
+    FROM debian:buster-slim
+    
+    COPY +build/example-rust example-rust
+    
+    RUN apt-get update -y \
+        && apt-get install -y --no-install-recommends openssl libcurl4 ca-certificates \
+        # Clean up
+        && apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+
+    EXPOSE 9091
+    ENTRYPOINT ["./example-rust"]
+    SAVE IMAGE example-rust:latest # use SAVE IMAGE --push myuser/myrepo:latest to push to Docker Hub
```

We can push this to our registry.

```suggestion
    SAVE IMAGE --push earthly/examples:rust
```